tests/BuildScriptGenerator.Tests/DotnetCore/GlobalJsonSdkResolverTest.cs

```diff
@@ -0,0 +1,1218 @@
+            {
+                Version = "3.1.100",
+                RollForward = GlobalJsonSdkResolver.Feature,
+            };
+            var globalJsonHelper = GetGlobalJsonHelper();
+
+            // Act
+            var actual = globalJsonHelper.GetSatisfyingSdkVersion(globalJson, availableVersions);
+
+            // Assert
+            Assert.Equal(expectedVersion, actual);
+        }
+
+        [Fact]
+        public void Feature_ReturnsNull_IfNextHigherFeatureAndLatestPatchLevelIsNotPresent()
+        {
+            // Arrange
+            var availableVersions = new[]
+            {
+                "2.1.100",
+                "3.0.100",
+                "3.0.101",
+                "3.1.200",
+                "3.1.201",
+            };
+            var globalJson = new GlobalJsonModel();
+            globalJson.Sdk = new Sdk
+            {
+                Version = "3.0.102",
+                RollForward = GlobalJsonSdkResolver.Feature,
+            };
+            var globalJsonHelper = GetGlobalJsonHelper();
+
+            // Act
+            var actual = globalJsonHelper.GetSatisfyingSdkVersion(globalJson, availableVersions);
+
+            // Assert
+            Assert.Null(actual);
+        }
+
+        [Fact]
+        public void Minor_ReturnsLatestPatchLevel_IfSpecifiedMinorAndFeatureArePresent()
+        {
+            // Arrange
+            var expectedVersion = "3.1.101";
+            var availableVersions = new[] { "2.1.100", "3.1.100", "3.1.101" };
+            var globalJson = new GlobalJsonModel();
+            globalJson.Sdk = new Sdk
+            {
+                Version = "3.1.100",
+                RollForward = GlobalJsonSdkResolver.Minor,
+            };
+            var globalJsonHelper = GetGlobalJsonHelper();
+
+            // Act
+            var actual = globalJsonHelper.GetSatisfyingSdkVersion(globalJson, availableVersions);
+
+            // Assert
+            Assert.Equal(expectedVersion, actual);
+        }
+
+        [Fact]
+        public void Minor_ReturnsNextHigherFeatureAndLatestPatchLevel_IfSpecifiedMinorIsPresentAndNotFeature()
+        {
+            // Arrange
+            var expectedVersion = "3.1.201";
+            var availableVersions = new[]
+            {
+                "2.1.100",
+                "3.1.200",
+                "3.1.201"
+            };
+            var globalJson = new GlobalJsonModel();
+            globalJson.Sdk = new Sdk
+            {
+                Version = "3.1.100",
+                RollForward = GlobalJsonSdkResolver.Minor,
+            };
+            var globalJsonHelper = GetGlobalJsonHelper();
+
+            // Act
+            var actual = globalJsonHelper.GetSatisfyingSdkVersion(globalJson, availableVersions);
+
+            // Assert
+            Assert.Equal(expectedVersion, actual);
+        }
+
+        [Fact]
+        public void Minor_ReturnsNextMinorAndNextHigherFeatureAndLatestPatchLevel_IfSpecifiedMinorIsNotPresent()
```

Comment:
yes this is the test i was looking for? is it going to break the user? minor version releases breaking changes as well right?

Reply:
Few things to consider here:
- It will be highly unlikely that a specified version would not exist, like we would having a bunch of sdks in our storage.
- .NET Core team would not do breaking changes in minor versions intentionally. As I mentioned in one of my earlier comments, if there is no global.json spec the default is `latest major` so what I am suggesting we do is `>=minor` which is far safer option.